SmartDeviceLink/SDLFileManager.m

```diff
@@ -403,6 +402,72 @@ - (void)sdl_uploadFile:(SDLFile *)file completionHandler:(nullable SDLFileManage
 }
 
+#pragma mark Artworks
+
+- (void)uploadArtwork:(SDLArtwork *)artwork completionHandler:(nullable SDLFileManagerUploadArtworkCompletionHandler)completion {
+    [self uploadFile:artwork completionHandler:^(BOOL success, NSUInteger bytesAvailable, NSError * _Nullable error) {
+        if (completion == nil) { return; }
+        if ([self isErrorACannotOverwriteError:error]) {
+            // Artwork with same name already uploaded to remote
+            return completion(true, artwork.name, bytesAvailable, nil);
+        }
+        completion(success, artwork.name, bytesAvailable, error);
+    }];
+}
+
+- (void)uploadArtworks:(NSArray<SDLArtwork *> *)artworks completionHandler:(nullable SDLFileManagerMultiUploadArtworkCompletionHandler)completion {
+    [self uploadArtworks:artworks progressHandler:nil completionHandler:completion];
+}
+
+- (void)uploadArtworks:(NSArray<SDLArtwork *> *)artworks progressHandler:(nullable SDLFileManagerMultiUploadArtworkProgressHandler)progressHandler completionHandler:(nullable SDLFileManagerMultiUploadArtworkCompletionHandler)completion {
+    if (artworks.count == 0) {
+        @throw [NSException sdl_missingFilesException];
+    }
+
+    [self uploadFiles:artworks progressHandler:^BOOL(SDLFileName * _Nonnull fileName, float uploadPercentage, NSError * _Nullable error) {
+        if (progressHandler == nil) { return YES; }
+        if ([self isErrorACannotOverwriteError:error]) {
+            // Artwork with same name already uploaded to remote
+            return progressHandler(fileName, uploadPercentage, nil);
+        }
+        return progressHandler(fileName, uploadPercentage, error);
+    } completionHandler:^(NSError * _Nullable error) {
+        if (completion == nil) { return; }
+
+        NSMutableSet<NSString *> *successfulArtworkUploadNames = [NSMutableSet set];
+        for (SDLArtwork *artwork in artworks) {
+            [successfulArtworkUploadNames addObject:artwork.name];
+        }
+        NSMutableDictionary *unsuccessfulArtworkUploadErrorUserInfo = [[NSMutableDictionary alloc] initWithDictionary:error.userInfo];
+
+        if (error != nil) {
+            for (NSString *erroredArtworkName in error.userInfo) {
+                if (![self isErrorACannotOverwriteError:[error.userInfo objectForKey:erroredArtworkName]]) {
+                    [successfulArtworkUploadNames removeObject:erroredArtworkName];
+                } else {
+                    // An overwrite error means that the artwork is already uploaded to the remote
+                    [unsuccessfulArtworkUploadErrorUserInfo removeObjectForKey:erroredArtworkName];
+                }
+            }
+        }
+
+        return completion([NSArray arrayWithArray:[successfulArtworkUploadNames allObjects]], unsuccessfulArtworkUploadErrorUserInfo == nil || unsuccessfulArtworkUploadErrorUserInfo.count == 0 ? nil : [[NSError alloc] initWithDomain:error.domain code:error.code userInfo:unsuccessfulArtworkUploadErrorUserInfo]);
```

Comment:
You only have to check count here, you don't have to check nil first.

Reply:
Fixed